app/_src/policies/meshtimeout.md

```diff
@@ -0,0 +1,315 @@
+---
+title: Mesh Timeout (beta)
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [Timeout policy](../timeout).
+{% endwarning %}
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅         | ✅   | ✅    |
+| MeshSubset        | ✅         | ❌   | ❌    |
+| MeshService       | ✅         | ✅   | ❌    |
+| MeshServiceSubset | ✅         | ❌   | ❌    |
+| MeshGatewayRoute  | ❌         | ❌   | ❌    |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+This policy enables {{site.mesh_product_name}} to set timeouts on the inbound and outbound connections depending on the protocol. Using this policy you can configure TCP and HTTP timeouts. Timeout configuration is split into two sections: common configuration and http configuration. Common config is applied to both HTTP and TCP communication.
+
+MeshTimeout policy lets you configure multiple timeouts:
+
+- connectionTimeout
+- idleTimeout
+- http requestTimeout
+- http streamIdleTimeout
+- http maxStreamDuration
+- http maxConnectionDuration
+
+### Timeouts explained
```

Comment:
I think we can borrow from [here](https://kuma.io/docs/2.0.x/policies/timeout/#configuration) a more structured view, mentioning at once the default value and place in the Envoy config

Reply:
Maybe it can be a table

Reply:
I am not sure, I wanted this to be just explanation of how those timeouts work, a more structured view in my opinion should be in reference docs, which we don't have now for new policies. As for the envoy config I don't feel like those internals should be described here. Envoy docs is for this

Reply:
The reference doc is going to be auto-generated, I'm not sure how it's going to work, but I guess we won't be able to put additional stuff there.

About Envoy confs, okay let's not add Envoy refs, we can always do this in the future if we need.